docs/libcurl/opts/CURLOPT_SSL_OPTIONS.md

```diff
@@ -86,6 +86,13 @@ certificate that supports client authentication in the OS certificate store it
 could be a privacy violation and unexpected.
 (Added in 7.77.0)
 
+## CURLSSLOPT_EARLYDATA
+
+Tell libcurl to try sending application data as TLS1.3 early data. This option
+is only supported for OpenSSL, WolfSSL and GnuTLS. This option only works when
+using HTTP/1.1 without a proxy otherwise its a no-op.
```

Comment:
What happens if this flag is set and the TLS negotiation ends up with a lower version than 1.2? It would be good to mention that here.

Also, since QUIC uses TLS 1.3 internally, it might be good to explicitly mention if that is also covered by this flag or not.

Reply:
> What happens if this flag is set and the TLS negotiation ends up with a lower version than 1.3

in this case, or when the server sends a `HelloRetryRequest`, openssl and others either raise an error or let us know that early data was rejected. We resend the request again in those cases post-handshake, since early data was not sent.

Should we mention that this option only works on a best effort basis, and in situations when it wasn't possible to send early data we fallback to resending the request post handshake?

We didn't implement early data for QUIC